src/openrct2-ui/interface/ViewportInteraction.cpp

```diff
@@ -391,34 +409,37 @@ int32_t viewport_interaction_get_item_right(const ScreenCoordsXY& screenCoords,
             banner->FormatTextTo(ft, /*addColour*/ true);
             ft.Add<rct_string_id>(STR_MAP_TOOLTIP_STRINGID_CLICK_TO_MODIFY);
             ft.Add<rct_string_id>(sceneryEntry->name);
-            return info->SpriteType;
+            return info;
         }
+        default:
+            break;
```

Comment:
Why did the `default` become neccessary? I'd rather we handled the extra cases explicitly, default might be painful in the future if we add more values to the enum

Reply:
its a bit of an annoyance but this part of the function is split into two switches with some middle code and if it is not in the first switch then it will be in the second switch.

Reply:
Ok